### DISCUSSION TITLE
Constraints for predictions

### DISCUSSION THREAD

**Discussion by johnf1004** [ourownstory/neural_prophet#151] [Q&A - get help using NeuralProphet] [answered]
Hi,

It looks like for now, constraints such as non-negativity cannot be imposed for predictions/forecasts (correct me if I'm wrong!) I guess this would be achieved through giving the user a hyperparameter option for the activation function of the network? Is this something that is coming in future versions?

FWIW the data I'm using is 0 at many/most days, and >>0 roughly twice a week. Imagine something like lottery ticket sales where the lottery occurs twice a week. So any other tips for working with a series like this are greatly appreciated. 

John

**Comment by johnf1004** [comment]
Probably related to #141

**Comment by ourownstory** [comment]
Hi John, 
Thank you for bringing this up. 
Let's further discuss this in issue #141.

**Comment by ourownstory** [comment, marked as answer]
Hi John,

Regarding your problem at hand, I would advise considering the following options:
* use "minmax" normalization (to keep zeros zero) `NeuralProphet(normalize_y="minmax")`
* consider modelling it as a weekly series (by downsampling your data in preprocessing)
* if modelling it as a weekly series is not possible, you can set the n_lags and n_forecasts to multiples of 7. Additionally you could add different rolling averages as lagged regressors, e.g.

```python
df['A'] = df['y'].rolling(14, min_periods=1).mean()
df['B'] = df['y'].rolling(56, min_periods=1).mean()
m = m.add_lagged_regressor(name='A', only_last_value=True)
m = m.add_lagged_regressor(name='B', only_last_value=True)
```
If you still encounter negative forecasts after trying this, there might be an issue - please feel free to open a new issue then.

**Comment by johnf1004** [comment]
Thanks so much for the suggestions - I will give them a try.

**Comment by ourownstory** [comment]
Great, please let us know if it worked for you!

**Comment by boozesounds** [comment]
Adding a rolling average has been very helpful. Thank you.
